aws_batch/postprocess.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
+import os
+import sys
+import multiprocessing
+import subprocess
+import json
+import csv
+import shelve
+import argparse
+import re
+import time
+import random
+import datetime
+import gzip
+import logging
+import math
+
+KEY_S3_PATH = 's3://czbiohub-infectious-disease/idseq-alpha.pem'
+ROOT_DIR = '/mnt'
+DEST_DIR = ROOT_DIR + '/idseq/data' # generated data go here
+REF_DIR  = ROOT_DIR + '/idseq/ref' # referene genome / ref databases go here
+
+# Global variable examples, to be overwritten by environment variables
+INPUT_BUCKET = 's3://czbiohub-idseq-samples-development/samples/3/60/results'
+OUTPUT_BUCKET = 's3://czbiohub-idseq-samples-development/samples/3/60/postprocess'
+
+# references
+ACCESSION2TAXID = 's3://czbiohub-infectious-disease/references/accession2taxid.db.gz'
+LINEAGE_SHELF = 's3://czbiohub-infectious-disease/references/taxid-lineages.db'
+
+# input files
+ACCESSION_ANNOTATED_FASTA = 'taxids.rapsearch2.filter.deuterostomes.taxids.gsnapl.unmapped.bowtie2.lzw.cdhitdup.priceseqfilter.unmapped.star.fasta'
+
+# output files
+TAXID_ANNOT_FASTA = 'taxid_annot.fasta'
+TAXID_ANNOT_SORTED_FASTA_NT = 'taxid_annot_sorted_nt.fasta'
+TAXID_LOCATIONS_JSON_NT = 'taxid_locations_nt.json'
+TAXID_ANNOT_SORTED_FASTA_NR = 'taxid_annot_sorted_nr.fasta'
+TAXID_LOCATIONS_JSON_NR = 'taxid_locations_nr.json'
+LOGS_OUT_BASENAME = 'postprocess-log'
+
+# convenience functions
+def return_merged_dict(dict1, dict2):
+    result = dict1.copy()
+    result.update(dict2)
+    return result
+
+# processing functions
+def accession2taxid(read_id, accession2taxid_dict, hit_type, lineage_map):
+    accid_short = ((read_id.split(hit_type+':'))[1].split(":")[0]).split(".")[0]
+    taxid = accession2taxid_dict.get(accid_short, "NA")
+    species_taxid, genus_taxid, family_taxid = lineage_map.get(taxid, ("-1", "-2", "-3"))
```

Comment:
Isn't this supposed to be -100, -200, -300?

Reply:
yes